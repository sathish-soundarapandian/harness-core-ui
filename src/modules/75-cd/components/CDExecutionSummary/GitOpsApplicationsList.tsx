/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import cx from 'classnames'
import { Link, useParams } from 'react-router-dom'
import { Popover, Position } from '@blueprintjs/core'
import { Icon } from '@harness/icons'
import { Layout, Text } from '@harness/uicore'
import { String } from 'framework/strings'
import type { Application } from 'services/cd-ng'
import routes from '@common/RouteDefinitions'
import type { ProjectPathProps, ModulePathParams } from '@common/interfaces/RouteInterfaces'
import css from './CDExecutionSummary.module.scss'

interface GitOpsApplicationsListProps {
  applications: Application[]
  className?: string
  limit?: number
}

export function GitOpsApplicationsList({
  applications,
  limit = 2,
  className
}: GitOpsApplicationsListProps): React.ReactElement {
  const { orgIdentifier, projectIdentifier, accountId, module } = useParams<ProjectPathProps & ModulePathParams>()
  return (
    <div className={cx(css.main, className)}>
      <>
        <div className={css.gitOpsApps}>
          <Icon name="gitops-application" className={css.appIcon} size={14} />
          <Text>
            {applications.slice(0, limit).map((app: Application, index: number) => {
              return (
                <Link
                  key={index}
                  onClick={e => e.stopPropagation()}
                  to={routes.toGitOpsApplication({
                    orgIdentifier,
                    projectIdentifier,
                    accountId,
                    module,
                    applicationId: (app.identifier || app.name) as string,
                    agentId: app.agentIdentifier
                  })}
                >
                  <Text>{app.name}</Text>
                </Link>
              )
            })}
          </Text>
        </div>
        {applications.length > limit ? (
          <>
            ,&nbsp;
            <Popover
              wrapperTagName="div"
              targetTagName="div"
              interactionKind="hover"
              position={Position.RIGHT}
              className={css.serviceWrapper}
            >
              <String
                tagName="div"
                className={cx(css.serviceName, css.count)}
                stringID={'common.plusNumberNoSpace'}
                vars={{ number: Math.abs(applications.length - limit) }}
              />
              <Layout.Vertical padding="small">
                {applications.slice(limit).map((app, index) => (
                  <Link
                    key={index}
                    onClick={e => e.stopPropagation()}
                    to={routes.toGitOpsApplication({
                      orgIdentifier,
                      projectIdentifier,
                      accountId,
                      module,
                      applicationId: (app.identifier || app.name) as string,
                      agentId: app.agentIdentifier
                    })}
                  >
                    <Text>{app.name}</Text>
                  </Link>
                ))}
              </Layout.Vertical>
            </Popover>
          </>
        ) : null}
      </>
    </div>
  )
}
