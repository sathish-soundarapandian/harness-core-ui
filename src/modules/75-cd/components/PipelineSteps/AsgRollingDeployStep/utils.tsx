/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import * as Yup from 'yup'
import { getMultiTypeFromValue, MultiTypeInputType } from '@harness/uicore'
import type { UseStringsReturn } from 'framework/strings'

export const minimumHealthyPercentageSchema = (
  getString: UseStringsReturn['getString']
): Yup.Schema<number | undefined> => {
  return Yup.lazy((value: number): any => {
    /* istanbul ignore else */
    if (getMultiTypeFromValue(value as any) === MultiTypeInputType.FIXED) {
      return Yup.mixed().test({
        test(): boolean | Yup.ValidationError {
          if (!value) {
            return true
          }
          if (value > 100) {
            return this.createError({ message: getString('cd.minimumHealthyPercentageMaxLimit') })
          }
          if (value < 0) {
            return this.createError({ message: getString('cd.minimumHealthyPercentageMinLimit') })
          }
          return true
        }
      })
    }
    return Yup.number()
  })
}

export const instanceWarmupSchema = (getString: UseStringsReturn['getString']): Yup.Schema<number | undefined> => {
  return Yup.lazy((value: number): any => {
    /* istanbul ignore else */
    if (getMultiTypeFromValue(value as any) === MultiTypeInputType.FIXED) {
      return Yup.mixed().test({
        test(): boolean | Yup.ValidationError {
          if (!value) {
            return true
          }
          if (value < 0) {
            return this.createError({ message: getString('cd.instanceWarmupError') })
          }
          return true
        }
      })
    }
    return Yup.number()
  })
}
