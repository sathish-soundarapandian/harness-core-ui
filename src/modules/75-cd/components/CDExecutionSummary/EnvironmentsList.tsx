/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { Link, useParams } from 'react-router-dom'
import { Popover, Position } from '@blueprintjs/core'
import { Icon, Layout, Text, Container } from '@harness/uicore'
import { defaultTo } from 'lodash-es'
import { FontVariation } from '@harness/design-system'
import cx from 'classnames'

import { getScopeFromValue } from '@common/components/EntityReference/EntityReference'
import { getIdentifierFromScopedRef } from '@common/utils/utils'
import routes from '@common/RouteDefinitions'

import { Scope } from '@common/interfaces/SecretsInterface'
import type { ProjectPathProps, ModulePathParams } from '@common/interfaces/RouteInterfaces'
import { String } from 'framework/strings'
import css from './CDExecutionSummary.module.scss'

interface EnvironmentsListProps {
  environments: string[]
  className?: string
  limit?: number
}

export function EnvironmentsList({ environments, limit = 2, className }: EnvironmentsListProps): React.ReactElement {
  const { orgIdentifier, projectIdentifier, accountId, module } = useParams<ProjectPathProps & ModulePathParams>()

  return (
    <div className={cx(css.main, className)}>
      {environments.length > 0 ? (
        <>
          <div className={css.environments}>
            <Icon name="environments" className={css.envIcon} size={14} />
            <Container flex>
              {environments.slice(0, limit).map(env => {
                const envScope = getScopeFromValue(defaultTo(env, ''))
                return (
                  <Text className={css.envName} lineClamp={1} key={env} margin={{ right: 'small' }}>
                    <Link
                      to={`${routes.toEnvironmentDetails({
                        accountId,
                        ...(envScope != Scope.ACCOUNT && { orgIdentifier: orgIdentifier }),
                        ...(envScope === Scope.PROJECT && { projectIdentifier: projectIdentifier }),
                        environmentIdentifier: defaultTo(getIdentifierFromScopedRef(env), ''),
                        module,
                        accountRoutePlacement: 'settings'
                      })}`}
                    >
                      {env}
                    </Link>
                  </Text>
                )
              })}
            </Container>
          </div>
          {environments.length > limit ? (
            <>
              ,&nbsp;
              <Popover
                wrapperTagName="div"
                targetTagName="div"
                interactionKind="hover"
                position={Position.RIGHT}
                className={css.serviceWrapper}
              >
                <String
                  tagName="div"
                  className={cx(css.serviceName, css.count)}
                  stringID={'common.plusNumberNoSpace'}
                  vars={{ number: Math.abs(environments.length - limit) }}
                />
                <Layout.Vertical padding="small">
                  {environments.slice(limit).map((environment, index) => (
                    <Text
                      className={css.executionItemDetails}
                      font={{ variation: FontVariation.FORM_LABEL }}
                      key={index}
                    >
                      {environment}
                    </Text>
                  ))}
                </Layout.Vertical>
              </Popover>
            </>
          ) : null}
        </>
      ) : null}
    </div>
  )
}
