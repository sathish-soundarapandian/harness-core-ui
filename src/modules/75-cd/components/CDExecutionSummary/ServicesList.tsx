/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { Link, useParams } from 'react-router-dom'
import { defaultTo } from 'lodash-es'

import { Popover, Position } from '@blueprintjs/core'
import { Icon, Text } from '@harness/uicore'

import cx from 'classnames'

import { String } from 'framework/strings'
import type { ServiceExecutionSummary } from 'services/cd-ng'
import { getScopeFromValue } from '@common/components/EntityReference/EntityReference'
import { getIdentifierFromScopedRef } from '@common/utils/utils'
import routes from '@common/RouteDefinitions'

import { Scope } from '@common/interfaces/SecretsInterface'
import type { ProjectPathProps, ModulePathParams } from '@common/interfaces/RouteInterfaces'

import { ServicePopoverCard } from '../ServicePopoverCard/ServicePopoverCard'
import { ServicesTable } from './ServicesTable'
import css from './CDExecutionSummary.module.scss'

interface ServicesListProps {
  services: ServiceExecutionSummary[]
  className?: string
  limit: number
}

export function ServicesList({ services, limit = 2, className }: ServicesListProps): React.ReactElement {
  const { orgIdentifier, projectIdentifier, accountId, module } = useParams<ProjectPathProps & ModulePathParams>()

  return (
    <div className={cx(css.main, className)}>
      <Icon name="services" className={css.servicesIcon} size={18} />
      <div className={css.servicesList}>
        {services.slice(0, limit).map(service => {
          const { identifier } = service
          const serviceScope = getScopeFromValue(defaultTo(identifier, ''))

          return (
            <Popover
              key={identifier}
              wrapperTagName="div"
              targetTagName="div"
              interactionKind="hover"
              position={Position.BOTTOM_RIGHT}
              className={css.serviceWrapper}
            >
              <Text className={css.serviceName} lineClamp={1} color="grey800">
                <Link
                  to={`${routes.toServiceStudio({
                    accountId,
                    ...(serviceScope != Scope.ACCOUNT && { orgIdentifier: orgIdentifier }),
                    ...(serviceScope === Scope.PROJECT && { projectIdentifier: projectIdentifier }),
                    serviceId: getIdentifierFromScopedRef(defaultTo(identifier, '')),
                    module,
                    accountRoutePlacement: 'settings'
                  })}`}
                >
                  {service.displayName}
                </Link>
              </Text>
              <ServicePopoverCard service={service} />
            </Popover>
          )
        })}
        {services.length > limit ? (
          <Popover
            wrapperTagName="div"
            targetTagName="div"
            interactionKind="hover"
            position={Position.BOTTOM_RIGHT}
            className={css.serviceWrapper}
          >
            <String
              tagName="div"
              className={cx(css.serviceName, css.count)}
              stringID={'common.plusNumberNoSpace'}
              vars={{ number: Math.abs(services.length - limit) }}
            />
            <ServicesTable services={services.slice(limit)} />
          </Popover>
        ) : null}
      </div>
    </div>
  )
}
