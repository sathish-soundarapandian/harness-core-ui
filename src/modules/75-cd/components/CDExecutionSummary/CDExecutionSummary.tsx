/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { Icon } from '@harness/uicore'
import { defaultTo, identity, uniqBy } from 'lodash-es'
import type {
  CDPipelineModuleInfo,
  CDStageModuleInfo,
  GitOpsExecutionSummary,
  ServiceExecutionSummary
} from 'services/cd-ng'
import type { ExecutionSummaryProps } from '@pipeline/factories/ExecutionFactory/types'

import { ServicesList } from './ServicesList'
import { EnvironmentsList } from './EnvironmentsList'
import css from './CDExecutionSummary.module.scss'

const LIMIT = 2

export function CDExecutionSummary(props: ExecutionSummaryProps<CDPipelineModuleInfo>): React.ReactElement {
  const { nodeMap, className } = props

  const { services, environments } = React.useMemo(() => {
    // eslint-disable-next-line @typescript-eslint/no-shadow
    const services: ServiceExecutionSummary[] = []
    // eslint-disable-next-line @typescript-eslint/no-shadow
    const environments: string[] = []

    nodeMap.forEach(stage => {
      const stageInfo = stage.moduleInfo?.cd || /* istanbul ignore next */ ({} as CDStageModuleInfo)
      // istanbul ignore else
      if (stageInfo.serviceInfo) {
        services.push(stageInfo.serviceInfo)
      }

      // This will removed with the multi service env list view effort
      const gitOpsEnvironments = Array.isArray(stageInfo.gitopsExecutionSummary?.environments)
        ? (stageInfo.gitopsExecutionSummary as Required<GitOpsExecutionSummary>).environments.map(envForGitOps =>
            defaultTo(envForGitOps.name, '')
          )
        : []

      // istanbul ignore else
      if (gitOpsEnvironments.length) {
        environments.push(...gitOpsEnvironments)
      } else if (stageInfo.infraExecutionSummary?.name || stageInfo.infraExecutionSummary?.identifier) {
        environments.push(stageInfo.infraExecutionSummary.name || stageInfo.infraExecutionSummary.identifier)
      }
    })

    return { services: uniqBy(services, s => s.identifier), environments: uniqBy(environments, identity) }
  }, [nodeMap])

  return (
    <div className={className ?? css.main}>
      <Icon name="cd-main" size={18} />
      <ServicesList services={services} limit={LIMIT} />
      <EnvironmentsList environments={environments} limit={LIMIT} />
    </div>
  )
}
