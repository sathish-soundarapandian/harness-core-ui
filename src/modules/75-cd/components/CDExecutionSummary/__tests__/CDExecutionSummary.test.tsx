/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render } from '@testing-library/react'
import type { ServiceExecutionSummary } from 'services/cd-ng'
import { TestWrapper } from '@common/utils/testUtils'

import { CDExecutionSummary } from '../CDExecutionSummary'
import { EnvironmentsList } from '../EnvironmentsList'
import { ServicesList } from '../ServicesList'
import { ServicesTableProps, ServicesTable } from '../ServicesTable'
import props from './props.json'

const environmentProps = {
  environments: ['demo1', 'demo2', 'demo3'],
  className: 'demo'
}

const servicesTableProps: ServicesTableProps = {
  services: [
    {
      identifier: 'nginx',
      displayName: 'todolist',
      deploymentType: 'Kubernetes',
      artifacts: {
        sidecars: [1],
        primary: []
      }
    },
    {
      identifier: 'service1',
      displayName: 'service1',
      deploymentType: 'Kubernetes'
    },
    {
      identifier: 'nginx',
      displayName: 'todolist',
      deploymentType: 'Kubernetes',
      artifacts: {
        primary: []
      }
    }
  ] as ServiceExecutionSummary[]
}

describe('<CDExecutionSummary /> tests', () => {
  test('snapshot test', () => {
    const { container } = render(
      <TestWrapper>
        <CDExecutionSummary {...(props as any)} />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })

  test('render environmentList', () => {
    const { container } = render(
      <TestWrapper>
        <EnvironmentsList {...environmentProps} />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })

  test('render serviceTable', () => {
    const { container } = render(
      <TestWrapper>
        <ServicesTable services={servicesTableProps.services} />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })

  test('render serviceList', () => {
    const { container } = render(
      <TestWrapper>
        <ServicesList services={servicesTableProps.services} limit={2} className={'demo'} />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })
})
