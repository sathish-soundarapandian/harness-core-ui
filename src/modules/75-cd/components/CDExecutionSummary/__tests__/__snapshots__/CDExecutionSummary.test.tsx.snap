// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CDExecutionSummary /> tests render environmentList 1`] = `
<div>
  <div
    class="main demo"
  >
    <div
      class="environments"
    >
      <span
        class="bp3-icon StyledProps--main envIcon"
        data-icon="environments"
      >
        <svg
          fill="none"
          height="14"
          viewBox="0 0 15 14"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.688 10.53a1.568 1.568 0 00-1.46 1.028h-2.2a1.893 1.893 0 00-1.263-1.262V9.07c1.767-.072 3.498-.631 3.498-1.713V1.803v-.072C11.263.541 9.226 0 7.315 0c-1.912 0-3.95.54-3.95 1.731v5.626c0 1.046 1.624 1.604 3.319 1.713v1.226a1.893 1.893 0 00-1.263 1.262h-2.38a1.569 1.569 0 100 1.082h2.38a1.875 1.875 0 003.607 0h2.2a1.569 1.569 0 101.46-2.11zM4.466 2.957c.775.325 1.839.487 2.867.487 1.027 0 2.091-.162 2.867-.487v1.497c-.127.198-1.1.667-2.867.667s-2.76-.469-2.867-.667V2.957zm0 4.364V5.716c.775.324 1.839.487 2.867.487 1.027 0 2.091-.163 2.867-.487V7.32c-.127.198-1.1.667-2.867.667s-2.76-.451-2.867-.65v-.017z"
            fill="currentColor"
          />
        </svg>
      </span>
      <div
        class="StyledProps--font StyledProps--main StyledProps--flex"
      >
        <p
          class="StyledProps--font Text--lineclamp StyledProps--main envName StyledProps--margin-right-small"
          style="--text-line-clamp: 1;"
        >
          <a
            href="/account/undefined/settings/resources/environments/demo1/details"
          >
            demo1
          </a>
        </p>
        <p
          class="StyledProps--font Text--lineclamp StyledProps--main envName StyledProps--margin-right-small"
          style="--text-line-clamp: 1;"
        >
          <a
            href="/account/undefined/settings/resources/environments/demo2/details"
          >
            demo2
          </a>
        </p>
      </div>
    </div>
    ,Â 
    <div
      class="bp3-popover-wrapper serviceWrapper"
    >
      <div
        class="bp3-popover-target"
      >
        <div
          class="serviceName count"
          tabindex="0"
        >
          common.plusNumberNoSpace
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<CDExecutionSummary /> tests render serviceList 1`] = `
<div>
  <div
    class="main demo"
  >
    <span
      class="bp3-icon StyledProps--main servicesIcon"
      data-icon="services"
    >
      <svg
        fill="currentColor"
        height="18"
        viewBox="0 0 25 24"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M10.52 2.991a1 1 0 011-1h2a1 1 0 011 1v1.241l.003.004c.004.005.011.01.022.013a7.953 7.953 0 012.019.837c.01.006.018.007.024.006h.005l.878-.879a1 1 0 011.414 0L20.3 5.627a1 1 0 010 1.415l-.879.878v.002l-.001.003c0 .006 0 .015.006.024.366.624.65 1.301.836 2.018.003.01.009.018.014.022a.015.015 0 00.003.002h1.242a1 1 0 011 1v2a1 1 0 01-1 1h-1.242l-.003.003a.042.042 0 00-.014.022 7.954 7.954 0 01-.838 2.02.041.041 0 00-.006.025v.004l.878.878a1 1 0 010 1.414l-1.414 1.414a1 1 0 01-1.414 0l-.879-.878h-.001l-.003-.001c-.006 0-.015 0-.025.006-.622.365-1.3.648-2.015.835a.043.043 0 00-.022.013l-.003.004v1.241a1 1 0 01-1 1h-2a1 1 0 01-1-1V19.75l-.002-.004a.042.042 0 00-.022-.013 7.956 7.956 0 01-2.017-.837.041.041 0 00-.025-.006l-.004.001-.878.879a1 1 0 01-1.415 0l-1.414-1.415a1 1 0 010-1.414l.878-.878.001-.004c0-.006 0-.016-.006-.025a7.955 7.955 0 01-.837-2.018.042.042 0 00-.013-.022.012.012 0 00-.004-.003H3.521a1 1 0 01-1-1v-2a1 1 0 011-1h1.241l.004-.002a.042.042 0 00.013-.022c.187-.717.47-1.394.836-2.018a.041.041 0 00.006-.024V7.92l-.88-.879a1 1 0 010-1.414l1.415-1.414a1 1 0 011.414 0l.878.879h.005c.006.001.015 0 .025-.005a7.955 7.955 0 012.018-.837.042.042 0 00.022-.014.015.015 0 00.002-.004v-1.24zm3 0h-2v1.241c0 .482-.337.872-.771.985a6.955 6.955 0 00-1.765.732c-.388.227-.902.19-1.243-.15l-.878-.879-1.414 1.414.879.88c.34.34.377.854.15 1.241a6.955 6.955 0 00-.731 1.764 1.024 1.024 0 01-.985.772H3.521v2h1.241c.482 0 .872.338.985.772a6.93 6.93 0 00.732 1.765c.227.387.19.902-.15 1.242l-.879.878 1.415 1.414.878-.878c.34-.34.855-.377 1.242-.15.545.32 1.137.568 1.764.731.434.113.771.503.771.985v1.241h2V19.75c0-.482.338-.872.772-.985a6.958 6.958 0 001.763-.73c.387-.227.901-.19 1.242.15l.878.88 1.414-1.415-.877-.877a1.024 1.024 0 01-.15-1.243c.32-.546.569-1.14.733-1.767.113-.434.503-.772.984-.772h1.242v-2h-1.242c-.481 0-.871-.337-.984-.772a6.952 6.952 0 00-.732-1.764 1.024 1.024 0 01.15-1.242l.88-.879-1.415-1.414-.878.879c-.34.34-.855.377-1.242.15a6.956 6.956 0 00-1.766-.732 1.024 1.024 0 01-.772-.985v-1.24zM12.477 7.53c-2.487 0-4.5 2.008-4.5 4.48 0 2.472 2.013 4.48 4.5 4.48 2.488 0 4.501-2.008 4.501-4.48 0-2.472-2.013-4.48-4.5-4.48zm-5.5 4.48c0-3.028 2.464-5.48 5.5-5.48 3.037 0 5.501 2.452 5.501 5.48 0 3.029-2.465 5.48-5.5 5.48-3.037 0-5.502-2.451-5.502-5.48z"
          fill-rule="evenodd"
        />
      </svg>
    </span>
    <div
      class="servicesList"
    >
      <div
        class="bp3-popover-wrapper serviceWrapper"
      >
        <div
          class="bp3-popover-target"
        >
          <p
            class="StyledProps--font Text--lineclamp StyledProps--main serviceName StyledProps--color StyledProps--color-grey800"
            style="--text-line-clamp: 1;"
            tabindex="0"
          >
            <a
              href="/account/undefined/settings/resources/services/nginx"
            >
              todolist
            </a>
          </p>
        </div>
      </div>
      <div
        class="bp3-popover-wrapper serviceWrapper"
      >
        <div
          class="bp3-popover-target"
        >
          <p
            class="StyledProps--font Text--lineclamp StyledProps--main serviceName StyledProps--color StyledProps--color-grey800"
            style="--text-line-clamp: 1;"
            tabindex="0"
          >
            <a
              href="/account/undefined/settings/resources/services/service1"
            >
              service1
            </a>
          </p>
        </div>
      </div>
      <div
        class="bp3-popover-wrapper serviceWrapper"
      >
        <div
          class="bp3-popover-target"
        >
          <div
            class="serviceName count"
            tabindex="0"
          >
            common.plusNumberNoSpace
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<CDExecutionSummary /> tests render serviceTable 1`] = `
<div>
  <table
    class="bp3-html-table bp3-small servicesTable"
  >
    <thead>
      <tr>
        <th>
          service
        </th>
        <th>
          primaryArtifactText
        </th>
        <th>
          sidecar
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          todolist
        </td>
        <td>
          artifactDisplay
        </td>
        <td>
          artifactDisplay
        </td>
      </tr>
      <tr>
        <td>
          service1
        </td>
        <td>
          -
        </td>
        <td>
          -
        </td>
      </tr>
      <tr>
        <td>
          todolist
        </td>
        <td>
          artifactDisplay
        </td>
        <td>
          -
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`<CDExecutionSummary /> tests snapshot test 1`] = `
<div>
  <div
    class="main"
  >
    <span
      class="bp3-icon StyledProps--main"
      data-icon="cd-main"
    >
      <svg
        height="18"
        viewBox="0 0 32 32"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M26.427 7.06A5.073 5.073 0 0129 11.52l-.077 8.934a5.073 5.073 0 01-2.537 4.35l-8.002 4.618a5.073 5.073 0 01-5.035.021l-7.776-4.402A5.073 5.073 0 013 20.583l.077-8.935a5.073 5.073 0 012.537-4.35l8.002-4.619a5.073 5.073 0 015.035-.02l7.776 4.402zm-2.83 6.938c-.42 1.586-2.029 2.538-3.601 2.141l-7.489-2.026a4.696 4.696 0 00-3.62.48 4.773 4.773 0 00-2.222 2.925 4.835 4.835 0 00.479 3.664 4.757 4.757 0 002.905 2.251 4.696 4.696 0 003.62-.48 4.773 4.773 0 002.222-2.924l.26-1.047a.25.25 0 00-.177-.302l-1.258-.34a.25.25 0 00-.307.18l-.256 1.025c-.426 1.602-2.064 2.556-3.651 2.127-1.582-.428-2.523-2.073-2.1-3.67.421-1.586 2.03-2.538 3.602-2.142l7.489 2.027a4.696 4.696 0 003.62-.48 4.773 4.773 0 002.222-2.925 4.835 4.835 0 00-.479-3.664 4.757 4.757 0 00-2.905-2.251 4.696 4.696 0 00-3.62.48 4.773 4.773 0 00-2.222 2.924l-.247.828a.25.25 0 00.174.312l1.259.34a.25.25 0 00.304-.169l.248-.827c.426-1.602 2.064-2.557 3.651-2.127 1.582.428 2.523 2.073 2.1 3.67z"
          fill="url(#cd-main_svg__a)"
          fill-rule="evenodd"
        />
        <defs>
          <lineargradient
            gradientUnits="userSpaceOnUse"
            id="cd-main_svg__a"
            x1="6.774"
            x2="21.5"
            y1="2"
            y2="30"
          >
            <stop
              stop-color="#62F91F"
            />
            <stop
              offset="1"
              stop-color="#45BD35"
            />
          </lineargradient>
        </defs>
      </svg>
    </span>
    <div
      class="main"
    >
      <span
        class="bp3-icon StyledProps--main servicesIcon"
        data-icon="services"
      >
        <svg
          fill="currentColor"
          height="18"
          viewBox="0 0 25 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M10.52 2.991a1 1 0 011-1h2a1 1 0 011 1v1.241l.003.004c.004.005.011.01.022.013a7.953 7.953 0 012.019.837c.01.006.018.007.024.006h.005l.878-.879a1 1 0 011.414 0L20.3 5.627a1 1 0 010 1.415l-.879.878v.002l-.001.003c0 .006 0 .015.006.024.366.624.65 1.301.836 2.018.003.01.009.018.014.022a.015.015 0 00.003.002h1.242a1 1 0 011 1v2a1 1 0 01-1 1h-1.242l-.003.003a.042.042 0 00-.014.022 7.954 7.954 0 01-.838 2.02.041.041 0 00-.006.025v.004l.878.878a1 1 0 010 1.414l-1.414 1.414a1 1 0 01-1.414 0l-.879-.878h-.001l-.003-.001c-.006 0-.015 0-.025.006-.622.365-1.3.648-2.015.835a.043.043 0 00-.022.013l-.003.004v1.241a1 1 0 01-1 1h-2a1 1 0 01-1-1V19.75l-.002-.004a.042.042 0 00-.022-.013 7.956 7.956 0 01-2.017-.837.041.041 0 00-.025-.006l-.004.001-.878.879a1 1 0 01-1.415 0l-1.414-1.415a1 1 0 010-1.414l.878-.878.001-.004c0-.006 0-.016-.006-.025a7.955 7.955 0 01-.837-2.018.042.042 0 00-.013-.022.012.012 0 00-.004-.003H3.521a1 1 0 01-1-1v-2a1 1 0 011-1h1.241l.004-.002a.042.042 0 00.013-.022c.187-.717.47-1.394.836-2.018a.041.041 0 00.006-.024V7.92l-.88-.879a1 1 0 010-1.414l1.415-1.414a1 1 0 011.414 0l.878.879h.005c.006.001.015 0 .025-.005a7.955 7.955 0 012.018-.837.042.042 0 00.022-.014.015.015 0 00.002-.004v-1.24zm3 0h-2v1.241c0 .482-.337.872-.771.985a6.955 6.955 0 00-1.765.732c-.388.227-.902.19-1.243-.15l-.878-.879-1.414 1.414.879.88c.34.34.377.854.15 1.241a6.955 6.955 0 00-.731 1.764 1.024 1.024 0 01-.985.772H3.521v2h1.241c.482 0 .872.338.985.772a6.93 6.93 0 00.732 1.765c.227.387.19.902-.15 1.242l-.879.878 1.415 1.414.878-.878c.34-.34.855-.377 1.242-.15.545.32 1.137.568 1.764.731.434.113.771.503.771.985v1.241h2V19.75c0-.482.338-.872.772-.985a6.958 6.958 0 001.763-.73c.387-.227.901-.19 1.242.15l.878.88 1.414-1.415-.877-.877a1.024 1.024 0 01-.15-1.243c.32-.546.569-1.14.733-1.767.113-.434.503-.772.984-.772h1.242v-2h-1.242c-.481 0-.871-.337-.984-.772a6.952 6.952 0 00-.732-1.764 1.024 1.024 0 01.15-1.242l.88-.879-1.415-1.414-.878.879c-.34.34-.855.377-1.242.15a6.956 6.956 0 00-1.766-.732 1.024 1.024 0 01-.772-.985v-1.24zM12.477 7.53c-2.487 0-4.5 2.008-4.5 4.48 0 2.472 2.013 4.48 4.5 4.48 2.488 0 4.501-2.008 4.501-4.48 0-2.472-2.013-4.48-4.5-4.48zm-5.5 4.48c0-3.028 2.464-5.48 5.5-5.48 3.037 0 5.501 2.452 5.501 5.48 0 3.029-2.465 5.48-5.5 5.48-3.037 0-5.502-2.451-5.502-5.48z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      <div
        class="servicesList"
      >
        <div
          class="bp3-popover-wrapper serviceWrapper"
        >
          <div
            class="bp3-popover-target"
          >
            <p
              class="StyledProps--font Text--lineclamp StyledProps--main serviceName StyledProps--color StyledProps--color-grey800"
              style="--text-line-clamp: 1;"
              tabindex="0"
            >
              <a
                href="/account/undefined/settings/resources/services/service1"
              >
                service1
              </a>
            </p>
          </div>
        </div>
        <div
          class="bp3-popover-wrapper serviceWrapper"
        >
          <div
            class="bp3-popover-target"
          >
            <p
              class="StyledProps--font Text--lineclamp StyledProps--main serviceName StyledProps--color StyledProps--color-grey800"
              style="--text-line-clamp: 1;"
              tabindex="0"
            >
              <a
                href="/account/undefined/settings/resources/services/nginx"
              >
                todolist
              </a>
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="main"
    >
      <div
        class="environments"
      >
        <span
          class="bp3-icon StyledProps--main envIcon"
          data-icon="environments"
        >
          <svg
            fill="none"
            height="14"
            viewBox="0 0 15 14"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.688 10.53a1.568 1.568 0 00-1.46 1.028h-2.2a1.893 1.893 0 00-1.263-1.262V9.07c1.767-.072 3.498-.631 3.498-1.713V1.803v-.072C11.263.541 9.226 0 7.315 0c-1.912 0-3.95.54-3.95 1.731v5.626c0 1.046 1.624 1.604 3.319 1.713v1.226a1.893 1.893 0 00-1.263 1.262h-2.38a1.569 1.569 0 100 1.082h2.38a1.875 1.875 0 003.607 0h2.2a1.569 1.569 0 101.46-2.11zM4.466 2.957c.775.325 1.839.487 2.867.487 1.027 0 2.091-.162 2.867-.487v1.497c-.127.198-1.1.667-2.867.667s-2.76-.469-2.867-.667V2.957zm0 4.364V5.716c.775.324 1.839.487 2.867.487 1.027 0 2.091-.163 2.867-.487V7.32c-.127.198-1.1.667-2.867.667s-2.76-.451-2.867-.65v-.017z"
              fill="currentColor"
            />
          </svg>
        </span>
        <div
          class="StyledProps--font StyledProps--main StyledProps--flex"
        >
          <p
            class="StyledProps--font Text--lineclamp StyledProps--main envName StyledProps--margin-right-small"
            style="--text-line-clamp: 1;"
          >
            <a
              href="/account/undefined/settings/resources/environments/UAT/details"
            >
              UAT
            </a>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
